src/main/scala/scala/compat/java8/converterImpl/StepsString.scala

```diff
@@ -0,0 +1,44 @@
+package scala.compat.java8.converterImpl
+
+import language.implicitConversions
+
+import scala.compat.java8.collectionImpl._
+import scala.compat.java8.runtime._
+
+import Stepper._
+
+////////////////////////////
+// Stepper implementation //
+////////////////////////////
+
+private[java8] class StepperStringCodePoint(underlying: String, var i0: Int, var iN: Int) extends IntStepper with EfficientSubstep {
+  def characteristics() = NonNull
+  def estimateSize = iN - i0
+  def hasNext = i0 < iN
+  def nextInt() = {
+    if (hasNext()) {
+      val cp = underlying.codePointAt(i0)
+      i0 += java.lang.Character.charCount(cp)
+      cp
+    }
+    else throwNSEE
+  }
+  def substep() = {
+    if (iN-3 > i0) {
+      var half = (i0+iN) >>> 1
+      if (java.lang.Character.isLowSurrogate(underlying.charAt(half))) half -= 1
+      val ans = new StepperStringCodePoint(underlying, i0, half)
+      i0 = half
+      ans
+    }
+    else null
+  }
+}
+
+/////////////////////////
+// Value class adapter //
+/////////////////////////
+
+final class RichStringCanStep(private val underlying: String) extends AnyVal with MakesIntStepper {
+  @inline def stepper: IntStepper with EfficientSubstep = new StepperStringCodePoint(underlying, 0, underlying.length)
```

Comment:
I would expect a method on `String` that is simply called `stepper` to return a Stepper over chars, not codepoints. Also, calling `.iterator` on a `String` now goes through the Stepper, so you get an Iterator over codepoints instead of chars.